src/nixenv.rs

```diff
@@ -0,0 +1,545 @@
+                        "Uninstalling path from the scratch profile due to conflicts"
+                    );
+
+                    {
+                        let output =
+                            tokio::process::Command::new(self.nix_store_path.join("bin/nix-env"))
+                                .process_group(0)
+                                .arg("--profile")
+                                .arg(&temporary_profile)
+                                .arg("--uninstall")
+                                .arg(root_path)
+                                .output()
+                                .await
+                                .map_err(|e| {
+                                    NixEnvError::StartNixCommand(
+                                        "nix-env --uninstall'ing a conflicting package",
+                                        e,
+                                    )
+                                })?;
+
+                        if !output.status.success() {
+                            return Err(NixEnvError::NixCommand(
+                                "nix-env --uninstall'ing a conflicting package",
+                                output,
+                            ));
+                        }
+                    }
+                }
+            }
+
+            let output = tokio::process::Command::new(self.nix_store_path.join("bin/nix-env"))
+                .process_group(0)
+                .arg("--profile")
+                .arg(&temporary_profile)
+                .arg("--install")
+                .arg(pkg)
+                .output()
+                .await
+                .map_err(|e| {
+                    NixEnvError::StartNixCommand("Adding a new package to the profile", e)
+                })?;
+
+            if !output.status.success() {
+                return Err(NixEnvError::AddPackage(pkg.to_path_buf(), output));
+            }
+        }
+
+        // Finish by setting the user provided profile to the new version we've constructed
+        {
+            let output = tokio::process::Command::new(self.nix_store_path.join("bin/nix-env"))
+                .process_group(0)
+                .arg("--profile")
+                .arg(self.profile)
+                .arg("--set")
+                .arg(&temporary_profile)
+                .output()
+                .await
+                .map_err(|e| {
+                    NixEnvError::StartNixCommand(
+                        "nix-env --profile ... --set ... the user's profile",
+                        e,
+                    )
+                })?;
+
+            if !output.status.success() {
+                return Err(NixEnvError::UpdateProfile(
+                    self.profile.to_path_buf(),
+                    output,
+                ));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct PackageInfo {
+    #[serde(default)]
+    outputs: HashMap<String, PathBuf>,
```

Comment:
You can't rely on only this:

```json5
{
  "elements": {
    "hello": {
      "active": true,
      "attrPath": "legacyPackages.x86_64-linux.hello",
      "originalUrl": "flake:nixpkgs",
      "outputs": null,
      "priority": 5,
      "storePaths": [
        "/nix/store/p09fxxwkdj69hk4mgddk4r3nassiryzc-hello-2.12.1"
      ],

```

Reply:
How did you get this output?

Reply:
nvm, I was looking at the new-style profile json